modules/c++/re/source/RegexSTL.cpp

```diff
@@ -0,0 +1,292 @@
+/* =========================================================================
+ * This file is part of re-c++
+ * =========================================================================
+ *
+ * (C) Copyright 2004 - 2014, MDA Information Systems LLC
+ *
+ * re-c++ is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation; either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this program; If not,
+ * see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#include "re/Regex.h" // this has to come before the #ifdef checks below
+
+// we use this file if we're not using actual PCRE itself
+#if defined(__CODA_CPP11)
+
+re::Regex::Regex(const std::string& pattern) :
+    mPattern(pattern)
+{
+    if (!mPattern.empty())
+    {
+        compile(mPattern);
+    }
+}
+
+void re::Regex::destroy()
+{
+}
+
+re::Regex::~Regex()
+{
+}
+
+re::Regex::Regex(const re::Regex& rhs)
+{
+    mPattern = rhs.mPattern;
+    compile(mPattern);
+}
+
+re::Regex& re::Regex::operator=(const re::Regex& rhs)
+{
+    if (this != &rhs)
+    {
+        mPattern = rhs.mPattern;
+
+        compile(mPattern);
+    }
+
+    return *this;
+}
+
+re::Regex& re::Regex::compile(const std::string& pattern)
+{
+    mPattern = replaceDot(pattern);
+    try 
```

Comment:
Rather than re-catching, just let the STL exception go (any caller that cares should be at some level in their application catching both CODA exceptions and STL exceptions).  Motivation is just for code simplicity / maintenance reasons - since the STL already behaves like we want, in general I try to avoid the extra couple lines for the re-catch unless we need to provide a specialized exception message.  So then update Regex.h's description to just say it throws in general rather than a specific exception that it throws.